sdk/src/message.rs

```diff
@@ -109,8 +109,17 @@ impl Message {
     }
 
     pub fn new(instructions: Vec<Instruction>) -> Self {
+        Self::new_with_payer(instructions, None)
+    }
+
+    pub fn new_with_payer(instructions: Vec<Instruction>, payer: Option<&Pubkey>) -> Self {
         let program_ids = get_program_ids(&instructions);
         let (mut signed_keys, unsigned_keys) = get_keys(&instructions);
+        if let Some(payer) = payer {
+            if signed_keys.is_empty() || signed_keys[0] != *payer {
```

Comment:
@mvines That doesn't look right. How about passing `payer` to `get_keys()` and if `Some`, make it the first key. Then you won't need any extra logic to check for duplicates. Looks like this code will end up with a duplicate key if any key after `signed_key[0]` is `payer`.

Reply:
That sounds better, follow-up coming up